pkg/network/protocols/http/ebpf_gotls.go

```diff
@@ -429,12 +478,23 @@ func (p *GoTLSProgram) detachHooks(probeIDs []manager.ProbeIdentificationPair) {
 	}
 }
 
-func (p *GoTLSProgram) Stop() {
-	if p == nil {
-		return
+func (p *GoTLSProgram) getHookedBinary(ino inodeNumber) *hookedBinary {
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+
+	bin, present := p.hookedBinaries[ino]
+	if !present {
+		return nil
 	}
 
-	close(p.procMonitor.done)
+	return bin
```

Comment:
I think that's redundant, as the default value should be nil in `ino` is missing

Reply:
Indeed!